src/main/java/spoon/support/reflect/declaration/CtMethodImpl.java

```diff
@@ -217,6 +219,35 @@ public boolean isShadow() {
 		return (CtMethod<T>) super.clone();
 	}
 
+	@Override
+	public Collection<CtMethod<?>> getTopDefinitions() {
+		List<CtMethod<?>> s = new ArrayList<>();
+
+		// first collect potential declarations of this method in the type hierarchy
+		ClassTypingContext context = new ClassTypingContext(this.getDeclaringType());
+		for (Object m : getDeclaringType().map(new AllTypeMembersFunction(CtMethod.class)).list()) {
```

Comment:
What about CtQuery#forEach instead of list() and then `for(m : list){}`? It will be shorter code, faster run and less memory needed.

Reply:
not possible since we need two passes on the list

Reply:
I created a PR on your PR here https://github.com/monperrus/spoon/pull/7